### PR TITLE
Remove crazy nested Unmarshal

### PR DESCRIPTION
**Pull request by soldiermoth**
& away we go

### CODE REVIEW

File: mpd/mpd.go [hunk 1]
```diff
@@ -126,6 +125,50 @@ type CommonAttributesAndElements struct {
...
+// to take care of the majority of our unmarshalling
+type wrappedAdaptationSet AdaptationSet
+
+// dtoAdaptationSet parses the items out of AdaptationSet
+// that give us trouble:
+// * slices of pointers
+// * Content Protection interface
+type dtoAdaptationSet struct {
+	wrappedAdaptationSet
+	Roles              []Role             `xml:"Role,omitempty"`
```
Comment (mjneil) [inline mpd/mpd.go]: I don't think slices of pointers are causing us any troubles, just the content protections. You should be able to remove `Roles` `Representations` and `AccesibilityElems` from here and let the default unmarshaling of `wrappedAdaptationSet` take care of them

**Comment by mjneil** [comment]
Also thank you for this, that huge nesting of the struct was such an eye sore

File: mpd/mpd.go (hunk 1 above)
Comment (soldiermoth) [reply mpd/mpd.go]: oh weird good catch. I swear they were giving me troubles earlier

File: mpd/mpd.go [hunk 2]
```diff
@@ -126,171 +125,80 @@ type CommonAttributesAndElements struct {
...
-			Group              *string               `xml:"group,attr"`
-			PAR                *string               `xml:"par,attr"`
-			MinBandwidth       *string               `xml:"minBandwidth,attr"`
-			MaxBandwidth       *string               `xml:"maxBandwidth,attr"`
-			MinWidth           *string               `xml:"minWidth,attr"`
-			MaxWidth           *string               `xml:"maxWidth,attr"`
-			MinHeight          *string               `xml:"minHeight,attr"`
-			MaxHeight          *string               `xml:"maxHeight,attr"`
-			ContentType        *string               `xml:"contentType,attr"`
-			ContentProtection  []ContentProtectioner `xml:"ContentProtection,omitempty"` // Common attribute, can be deprecated here
```
Comment (soldiermoth) [inline mpd/mpd.go]: REmoved this as it broke ordering & the XML to fail the XSD check

**Pull request merged**